modules/external_commands/inc/com/centreon/engine/modules/external_commands/processing.hh

```diff
@@ -233,12 +262,41 @@ namespace         modules {
 
         char* name(my_strtok(args, ";"));
         char* description(my_strtok(NULL, ";"));
-        service* svc(::find_service(name, description));
-        if (!svc)
+        std::pair<uint64_t, uint64_t> found_id(get_host_and_service_id(
+          name, description));
+        std::unordered_map<std::pair<uint64_t, uint64_t>,
+                           std::shared_ptr<service> >::const_iterator
+          found(configuration::applier::state::instance().services().find(found_id));
+
+        if (found != configuration::applier::state::instance().services().end() ||
+          found->second.get() != nullptr)
+          return ;
+        (*fptr)(found->second.get(), args + strlen(name) + strlen(description) + 2);
+      }
+
+      template <void (*fptr)(std::shared_ptr<service>, char*)>
+      static void _redirector_service_shared(
+        int id,
+        time_t entry_time,
+        char* args) {
+        (void)id;
+        (void)entry_time;
+
+        char* name(my_strtok(args, ";"));
+        char* description(my_strtok(NULL, ";"));
+        std::pair<uint64_t, uint64_t> found_id(get_host_and_service_id(
+          name, description));
+        std::unordered_map<std::pair<uint64_t, uint64_t>,
+                           std::shared_ptr<service> >::const_iterator
+          found(configuration::applier::state::instance().services().find(found_id));
+
+        if (found != configuration::applier::state::instance().services().end() ||
```

Comment:
Here, I don't have the original code but it looks similar...

Reply:
done